crates/common/tedge_config_macros/impl/src/query.rs

```diff
@@ -363,9 +381,26 @@ fn generate_string_writers(paths: &[VecDeque<&FieldOrGroup>]) -> TokenStream {
                 parse_quote_spanned! {ty.span()=>
                     WritableKey::#variant_name => self.#(#segments).* = None,
                 },
+                parse_quote_spanned! {ty.span()=>
+                    WritableKey::#variant_name => self.#(#segments).* = <#ty as AppendRemoveItem>::append(
+                        self.#(#segments).*.to_owned().or_else(|| #default_value),
+                        value
+                        .#parse
+                        .#convert_to_field_ty
+                        .map_err(|e| WriteError::ParseValue(Box::new(e)))?),
+                },
+                parse_quote_spanned! {ty.span()=>
+                    WritableKey::#variant_name => self.#(#segments).* = <#ty as AppendRemoveItem>::remove(
+                        self.#(#segments).*.to_owned().or_else(|| #default_value),
```

Comment:
Is there a reason why we try reading from the DTO in the first instance and then the reader if that's `None` rather than just using the value from the reader in any case?

Reply:
No, we can actually directly provide values from reader to append/remove functions. This is a leftover from previous implementation where I wanted to edit the dto value if it was `None` and provide it as `&mut` argument. However, in the mean time I changed the implementation and forgot to simplify it.